cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/monitor/RuntimeClientService.java

```diff
@@ -182,14 +203,35 @@ public void onProgramStopRequested(LongConsumer stopper) {
   /**
    * Creates an instance of {@link TopicRelayer} based on the topic.
    */
-  private TopicRelayer createTopicRelayer(CConfiguration cConf, String topic) {
+  private TopicRelayer createTopicRelayer(CConfiguration cConf,
+      String topic,
+      SchemaGenerator schemaGenerator,
+      DatumWriterFactory writerFactory,
+      DatumReaderFactory readerFactory) {
     TopicId topicId = NamespaceId.SYSTEM.topic(topic);
 
-    String programStatusTopic = cConf.get(Constants.AppFabric.PROGRAM_STATUS_EVENT_TOPIC);
+    String programStatusTopic = cConf.get(
+        Constants.AppFabric.PROGRAM_STATUS_EVENT_TOPIC);
     if (topic.matches("^" + Pattern.quote(programStatusTopic) + "[0-9]*$")) {
-      return new ProgramStatusTopicRelayer(topicId);
+      return new ProgramStatusTopicRelayer(topicId, pollTimeMillis);
     }
-    return new TopicRelayer(topicId);
+    String metricsTopic = cConf.get(Constants.Metrics.TOPIC_PREFIX);
+    // If metrics aggregation is enabled, we need a larger poll interval because
+    // polling at a faster rate than the metrics publishing rate will decrease
+    // the benefit of aggregation.
+    if (cConf.getBoolean(Constants.RuntimeMonitor.METRICS_AGGREGATION_ENABLED)
+        &&
+        topic.matches("^" + Pattern.quote(metricsTopic) + "[0-9" + "]]*$")) {
```

Comment:
Since the `metricsTopic` is a prefix (as suggested by the config name), I think using a simple start with is good enough rather than a regex.